tiledb/api/src/tests/examples/sparse_all.rs

```diff
@@ -0,0 +1,148 @@
+use std::rc::Rc;
+
+use crate::array::{
+    ArrayType, AttributeData, CellValNum, DimensionConstraints, DimensionData,
+    DomainData, SchemaData,
+};
+use crate::{physical_type_go, Datatype};
+
+pub type FnAcceptDimension = dyn Fn(&Parameters, Datatype) -> bool;
+pub type FnAcceptAttribute =
+    dyn Fn(&Parameters, Datatype, CellValNum, bool) -> bool;
+
+/// Configures construction of the `sparse_all` schema.
+#[derive(Clone)]
+pub struct Parameters {
+    /// Function which determines whether to add a dimension to the schema.
+    ///
+    /// By default, all types are added as dimensions except `Datatype::StringAscii`.
+    pub fn_accept_dimension: Rc<FnAcceptDimension>,
```

Comment:
Do we expect to modify those? Cant we use a Box type instead of reference counting?

Reply:
`Rc` is necessary for `#[derive(Clone)]`.
If we had `Box<dyn Thing>` then we can't clone it without requiring `Thing` to be `Clone`.
`Rc` seemed like a better choice.

Reply:
Is cloned essential here though? Can you explain why we need to clone?

Reply:
Cloning is not essential but it can be useful for a partial initialization, e.g.
```
let whatever = Parameters {
    fn_accept_dimension: Rc::new(fn_accept_uint64_only),
    ..Default::default()
};
```

Reply:
Oops, `Default::default()` is a poor example, because we don't actually need `Clone` for that.  Whatever the expression is, gets moved from; so if we are trying to mostly copy an existing set of parameters but change one field, then we want to have `Clone` for that to be convenient.
```
let whatever = Parameters {
    fn_accept_dimension: Rc::new(fn_accept_uint64_only),
    ..existing_parameters.clone()
};
```

This is a pattern we see in some of the proptest code, e.g.
```
                let domain_requirements = Rc::new(DomainRequirements {
                    array_type: Some(array_type),
                    ..domain_requirements.as_ref().clone()
                });
```

Reply:
Ok thanks for explaining. Resolving here.